torchkge/utils/training.py

```diff
@@ -53,7 +57,10 @@ def __len__(self):
         return get_n_batches(len(self.h), self.b_size)
 
     def __iter__(self):
-        return TrainDataLoaderIter(self)
```

Comment:
This should fill the attribute `self.iterator` with a call to `TrainDataLoaderIter(self)`. That way a new iteration on the dataset will relaunch a negative sampling step (c.f. the `TrainDataLoaderIter` initialization step).